### PR TITLE
Vr dependencies

### PR DESCRIPTION
**Pull request by wouterz**
Importing vr dependencies for oculus rift to work. The library is on git, as this seems to be the only way to make this work. Also did a little scene work to make it look better on the rift.

I had to fix alot of merge conflict with release as this branch has been open since last milestone, so please make sure i didnt remove anything by accident!

resolve issue [108](https://github.com/jessetilro/pixelperfect/issues/108)

### CODE REVIEW

**Comment by wouterz** [comment]
Travis failures

**Comment by wouterz** [comment]
All dependencies needed for using the rift are imported, But all files are stored local now. Jesse if you want you can throw this library on your repo and we can try to import them remotely once more, otherwise we will stick to this local library.

**Comment by jessetilro** [comment]
Wouter, will try to import the libraries remotely later, for now this is fine. Can you wait with merging until PR #110 is merged and pulled in your branch? I would like to inspect the effect on the test coverage.

**Comment by dmalarev** [comment]
Right now, the code looks good

File: src/main/java/nl/tudelft/pixelperfect/Game.java [hunk 1]
```diff
@@ -84,13 +178,30 @@ public Spaceship getSpaceship() {
...
+    }
+    if (moveBackwards) {
+      observer.move(VRApplication.getFinalObserverRotation().getRotationColumn(2).mult(-tpf * 8f));
+    }
+    if (rotateLeft) {
+      observer.rotate(0, 0.75f * tpf, 0);
+    }
+    if (rotateRight) {
+      observer.rotate(0, -0.75f * tpf, 0);
+    }
```
Comment (jessetilro) [inline src/main/java/nl/tudelft/pixelperfect/Game.java]: Why the movement, we don't allow the captain to move arround in the cockpit right?

File: pom.xml [hunk 2]
```diff
@@ -1,193 +1,226 @@
...
+
+	<properties>
+		<jme3.version>3.1.0-beta1</jme3.version>
+	</properties>
+
+	<repositories>
+		<repository>
+			<id>local</id>
+			<name>Local Repository</name>
+			<url>file://${basedir}/library</url>
```
Comment (jessetilro) [inline pom.xml]: It's a shame a lot of dependencies had to be included locally in order to make this work. We will need to find a solution to host (at least the majority of) the dependencies online.

**Comment by jessetilro** [comment]
Static analysis is clean and test coverage in all packages is ok except for the main package. (According to Cobertura the complexity in the main package has increased, so the branch coverage has dropped, but we were not planning on testing the main package anyway.) The system runs on my machine even without the VR device, that's great. LGTM.

File: src/main/java/nl/tudelft/pixelperfect/Game.java (hunk 1 above)
Comment (wouterz) [inline src/main/java/nl/tudelft/pixelperfect/Game.java]: I think moving around the cockpit might be needed. He might not be able to see everything from one stationary position.

File: pom.xml (hunk 2 above)
Comment (wouterz) [inline pom.xml]: Ideally yes, but after spending countless hours im just glad it works and would like to focus on the first playable spikes again for now

File: src/main/java/nl/tudelft/pixelperfect/Game.java (hunk 1 above)
Comment (jessetilro) [inline src/main/java/nl/tudelft/pixelperfect/Game.java]: This is not what our design documents specify, so I believe we must discuss this during the next meeting.

**Comment by dalderliesten** [comment]
Looks good to me.

File: src/main/java/nl/tudelft/pixelperfect/Spaceship.java [hunk 3]
```diff
@@ -122,6 +122,15 @@ public boolean isVictorious() {
   }
 
   /**
+   * Get the ship's timer.
+   * 
+   * @return The time the ship has been on its way.
+   */
+  public float getTimer() {
```
Comment (jessetilro) [inline src/main/java/nl/tudelft/pixelperfect/Spaceship.java]: This method is not tested.

**Pull request merged**